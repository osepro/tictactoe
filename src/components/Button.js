import React from "react";
import "../styles/button.css";

const Button = ({ play, position, canplay, winshown, boarditem }) => {
	const [a, b, c] = winshown;
	return (
		<div className="mainButton">
			<input
				type="button"
				onClick={play}
				data-item={position}
				value={boarditem[position].player}
				className="playButton"
				disabled={canplay}
				style={{
					background:
						position === a || position === b || position === c
							? "#00FF00"
							: "#000000",
				}}
			></input>
		</div>
	);
};

export default Button;
