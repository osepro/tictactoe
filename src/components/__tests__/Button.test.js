import React from "react";
import { shallow, mount } from "enzyme";
import Button from "components/Button";

const boarditem = [{ player: "", position: null }];
describe("button mount and snapshot", () => {
	let wrapped;
	beforeEach(() => {
		wrapped = mount(
			<Button
				position={0}
				play={(e) => {
					boarditem[0] = {
						player: "X",
						position: 0,
					};
				}}
				canplay={true}
				winshown={[]}
				boarditem={boarditem}
			/>
		);
	});
	afterEach(() => {
		wrapped.unmount();
	});

	it("contains input", () => {
		expect(wrapped).toMatchSnapshot();
	});
	it("contains button with value", () => {
		let eachPlay = wrapped.find("input");
		eachPlay.props().onClick();
		expect(wrapped.find(Button).prop("boarditem")).toEqual([
			{ player: "X", position: 0 },
		]);
	});
});

it("button click function called", () => {
	const btnClicked = jest.fn();
	const wrapped = shallow(
		<Button
			position={0}
			play={btnClicked}
			canplay={true}
			winshown={[]}
			boarditem={boarditem}
		/>
	);
	wrapped.find("input").simulate("click");
	expect(btnClicked).toHaveBeenCalled();
});
