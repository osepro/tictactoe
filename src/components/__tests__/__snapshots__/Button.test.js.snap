// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button mount and snapshot contains input 1`] = `ReactWrapper {}`;
